From 673d9a1e8c7d963c6258b3f471c1026bda702446 Mon Sep 17 00:00:00 2001
From: Javier Celaya <javier.celaya@flexvdi.com>
Date: Tue, 18 Apr 2017 10:50:52 +0200
Subject: [PATCH 1/5] Set errno in Windows too

Use WSASetLastError in Windows whenever errno is set in Unix platforms
---
 src/nopoll_conn.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/nopoll_conn.c b/src/nopoll_conn.c
index 2b91f67..c8ebd28 100644
--- a/src/nopoll_conn.c
+++ b/src/nopoll_conn.c
@@ -2286,6 +2286,8 @@ int         __nopoll_conn_receive  (noPollConn * conn, char  * buffer, int  maxl
 	/* memset (buffer, 0, maxlen * sizeof (char )); */
 #if defined(NOPOLL_OS_UNIX)
 	errno = 0;
+#elif defined(NOPOLL_OS_WIN32)
+	WSASetLastError(0);
 #endif
 	if ((nread = conn->receive (conn, buffer, maxlen)) == NOPOLL_SOCKET_ERROR) {
 		/* nopoll_log (conn->ctx, NOPOLL_LEVEL_DEBUG, " returning errno=%d (%s)", errno, strerror (errno)); */
@@ -3046,6 +3048,8 @@ noPollMsg   * nopoll_conn_get_msg (noPollConn * conn)
 		errno = NOPOLL_EWOULDBLOCK; /* simulate there is no data available to stop
 					       here. If there is no data indeed, on next
 					       call it will not fail */
+#elif defined(NOPOLL_OS_WIN32)
+		WSASetLastError(NOPOLL_EWOULDBLOCK); /* simulate there is no data available */
 #endif
 		return NULL;
 		
-- 
2.9.3

